velocity.js

```diff
@@ -525,29 +525,77 @@
 			return result;
 		}
 
+		/**
+		 * Shim for "fixing" IE's lack of support (IE < 9) for applying slice
+		 * on host objects like NamedNodeMap, NodeList, and HTMLCollection
+		 * (technically, since host objects have been implementation-dependent,
+		 * at least before ES2015, IE hasn't needed to work this way).
+		 * Also works on strings, fixes IE < 9 to allow an explicit undefined
+		 * for the 2nd argument (as in Firefox), and prevents errors when
+		 * called on other DOM objects.
+		 */
 		var _slice = (function() {
 			var slice = Array.prototype.slice;
 
 			try {
 				// Can't be used with DOM elements in IE < 9
 				slice.call(document.documentElement);
+				return Array.prototype.slice;
 			} catch (e) { // Fails in IE < 9
+
+				// Quick exit if we've already defined the function.
+				// Just return it now rather than redefining it.
+				if (Array.prototype._vel_slice !== undefined)
```

Comment:
This is an IIFE - called a single time and the results then stored in _slice on line 537 - it's not possible to call it again so no need to cache as it simply exists (or not)